tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLAccessTest.java

```diff
@@ -269,6 +269,11 @@ public void testXmlAccessWithLargerIndex() {
         BRunUtil.invoke(result, "testXmlAccessWithLargerIndex");
     }
 
+    @Test
+    public void testXMLNavigationWithUnionType() {
+        BRunUtil.invoke(navigation, "testXMLNavigationWithUnionType");
+    }
```

Comment:
Please also add some negative tests against access on unions (with and without xml members) that as a whole are not subtypes of xml.